api/src/main/java/org/openmrs/module/fhir2/api/translators/ToFhirProvenance.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators;
+
+/**
+ * Generic interface for a translator between OpenMRS data and FHIR provenance resource
+ *
+ * @param <T> OpenMRS data type
+ * @param <U> FHIR Provenance resource type
+ */
+public interface ToFhirProvenance<T, U> {
```

Comment:
I *think* since this interface is always expected to work on `Provenance` resources we could just make this `ToFhirProvenance<T>` and change the return types of the methods to be explicitly `Provenance`.